### ISSUE TITLE
[Feature] Sweep config: Specify parameter combinations as categorical

### ISSUE THREAD

**Issue by talesa** [wandb/wandb#1863]
**Is your feature request related to a problem? Please describe.**
Imagine a setting where you want to train a fully connected network but you have some space constraints - simplifying let's say you only want to train networks within some particular range of total number of hidden units (`hidden_units_min <= hidden_units <= hidden_units_max`).
`hidden_units` depends on `num_layers` and `layer_size`, `hidden_units = num_layers * layer_size`.
Naively we could run a sweep in a standard way, sweeping over all possibly combinations, but it’s very wasteful and slow because obviously `wandb` sweep doesn’t know about the relationship between `hidden_units`, `num_layers`, `layer_size`, and the `loss` which is a sum of performance metric (the lower the better) and a penalty on `hidden_units` deviating from the target hidden units range.
```
method: bayes
metric:
  goal: minimize
  name: loss  # loss = performance + penalty on hidden_units deviating from the target hidden_units range
parameters:
  num_layers:
    max: 4
    min: 1
    distribution: int_uniform
  layer_size:
    max: 128
    min: 8
    distribution: int_uniform
  ...
```
One way to address this problem is to “preprocess” the `sweep.yaml` and provide plausible combinations of pairs of `(num_layers, layer_size)` that satisfy your condition (which you can do because of your prior knowledge), which I'd imagine could yield a `sweep.yaml` along these lines (pending yaml syntax?)
```
parameters:
  "num_layers,layer_size":  # needs "..." to be interpreted as a string?
    values:
      - [1, 128]
      - [2, 64]
      - [3, 43]
      - [4, 32]
    distribution: categorical
  ...
```

**Describe the solution you’d like**
I would like to be able to specify `sweep.yaml` as per the suggestion above.

**Describe alternatives you’ve considered**
I have a hacky workaround: a new argument which takes a string, splits it by `,`, casts values to `int` and overwrites the other arguments `num_layers` and `layer_size`:
```
parser = argparse.ArgumentParser()
parser.add_argument("-lss", "--layer_size", help="Layer size", type=int, default=32)
parser.add_argument("-nl", "--num_layers", help="Number of layers", type=int, default=4)
parser.add_argument("-nllsss", "--num_layers_layer_size_sweep", help="For plausible architecture sweep purpose",
                    type=str, default=None)
args = parser.parse_args()

if args.num_layers_layer_size_sweep is not None:
    args.num_layers, args.layer_size = (int(i) for i in args.num_layers_layer_size_sweep.split(','))

wandb.init(config=args, ...)
```
and in `sweep.yaml`
```
parameters:
  num_layers_layer_size_sweep:
    values:
      - "4,32"
      - "3,48"
    distribution: categorical
  ...
```

**Comment by cvphelps** [comment]
Hey Adam thanks for the thoughtful suggestion. The hacky workaround looks right, and when we come back to Sweeps and prioritize new features, we will include this feature request in the list!

**Comment by rogelioamancisidor** [comment]
@cvphelps whats the status for this suggested feature? I would also like to specify which combinations in the `sweep.yaml` should be run.

**Comment by ramit-wandb** [comment]
Hi @rogelioamancisidor,

Conditional Parameter sweeps are still a planned feature and have not been implemented as of now. As of now, the original workaround suggested in this thread might be the best way to go to specify parameter combinations.

I'll let you know when I have some news regarding the update to this feature.

Thanks,
Ramit

**Comment by frederiknolte** [comment]
Is there any update on this? A major use case for me is that I am training models that weights that are specific for each seed. So if I want to train multiple seeds in one sweep, the paths must fit to the seeds.

**Comment by luisbergua** [comment]
Hey all! I just wanted to let you know that conditional sweeps is possible using Launch, check out [this](https://colab.research.google.com/drive/1WxLKaJlltThgZyhc7dcZhDQ6cjVQDfil#scrollTo=AFEzIxA6foC7) colab

**Comment by sydholl** [comment]
WandB Internal User commented: 
luisbergua commented: 
Hey all! I just wanted to let you know that conditional sweeps is possible using Launch, check out [this](https://colab.research.google.com/drive/1WxLKaJlltThgZyhc7dcZhDQ6cjVQDfil#scrollTo=AFEzIxA6foC7) colab
